### ISSUE TITLE
Using "{'$not': {'$size': 0}}" raises OperationFailure

### ISSUE THREAD

**Issue by ymost** [mongomock/mongomock#206]
The following code runs fine with pymongo:

```
col.insert_many([{'a': [1,2,3]}])
cursor = col.find({'a': {'$not': {'$size': 0}}})
cursor.next()
```

But with mongomock you get:

```
OperationFailure: BadValue $not needs a regex or a document
```

**Comment by dgarana** [comment]
Hi, this is related with this pull request.
https://github.com/mongomock/mongomock/pull/266

**Comment by pcorpet** [comment]
This was fixed by @srinivasreddy in #266, thanks Srinivas.
